### DISCUSSION TITLE
Welcome to Sigenergy-Home-Assistant-Integration Discussions!

### DISCUSSION THREAD

**Discussion by TypQxQ** [TypQxQ/Sigenergy-Home-Assistant-Integration#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

If need ask a question, search first and ask later in a new thread.

**Comment by jyfab** [comment]
Hi,

I have a Sigenstor EC 4.6 SP and tried to configure your integration on my Home Assistant setup, but I receive no data... Any idea how I can troubleshoot?
Thank you :)

**Reply by TypQxQ** [reply]
You can start by asking your installer for a screenshot of your modbus settings. Should look something like this:

![image](https://github.com/user-attachments/assets/b99c770b-f61f-4fe0-b78e-7caf5910f082)

**Reply by jyfab** [reply]
Hi,

My installer confirmed the port is 502 and the slave address is 1.
However I'm not sure about the IP address, what should I put in the secrets.yaml file?
I added the IP address of my inverter found on my router but I get the following errors:
Failed to connect [Errno 111] Connect call failed ('192.168.2.60', 502)
Pymodbus: Sigen: Error: device: 247 address: 30003 -> Modbus Error: [Connection] Not connected[AsyncModbusTcpClient 192.168.2.60:502]

Any idea how I can further troubleshoot?

thanks!

**Reply by EnterTheMaelstrom** [reply]
[edit] ooooh nevermind

**Reply by joostbijgithub** [reply]
> You can start by asking your installer for a screenshot of your modbus settings. Should look something like this:
> 
> ![image](https://private-user-images.githubusercontent.com/24956496/364732738-b99c770b-f61f-4fe0-b78e-7caf5910f082.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkxMTE2MDAsIm5iZiI6MTczOTExMTMwMCwicGF0aCI6Ii8yNDk1NjQ5Ni8zNjQ3MzI3MzgtYjk5Yzc3MGItZjYxZi00ZmUwLWI3OGUtN2NhZjU5MTBmMDgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjA5VDE0MjgyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUyODMzMDc1YWQzOTI5ZTNiYTM5OWRhNGViMjg2M2JkZjI2Zjg3NWJkODdjMTZjODIwYmIyYTZiMTE5MTg0NzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FW8Hj6nGTYGwVVnx1aaiqMjTa6RW546jUnIqPQmhXLA)

Sorry for the probably stupid question but does the ModBus Server Adress need to be filled in? My installer asks this question. There is no mention about it in the install guide (so my guess is not) but would like to be sure.  (I think that the Sig system needs to acts as the server, which is arranged by the last option. The first ones are if the Sig system acts as a client but that probably then not needed ...?

**Reply by TypQxQ** [reply]
I guess not but I don't know. This is what my installer sent to confirm it is active.
Your installer has access to more documentation than me.

**Comment by TypQxQ** [comment]
I used this software to find connection errors:
https://www.rilheva.com/rilheva-modbus-poll-desktop-edition/
I tried the web based version and couldn't connect but the downloadable one worked fine.

![image](https://github.com/user-attachments/assets/2787f28f-098c-4c24-b5a5-8fbff429ceb0)

Regards, Andrei

**Comment by HomerHH** [comment]
Thank you! 
I set it up today and it is working fine!

**Comment by jyfab** [comment]
Hi, so I queried again my installer and it turns out that modbus TCP was not enabled... Now that it is, I can fetch the data from my inverter :)

**Comment by EnterTheMaelstrom** [comment]
Hey,  
I just wanted to thank you for this easy integration of the modbus entries into HA.  
My system was activated yesterday and everything is working "out-of-the-box" perfectly fine. My installer was bamboozled about the seamless integration.  
I prepared everything beforehand and just entered the IP as soon as it popped up in my router and had all the data available before he told me to create a sigen account ;D
So thanks a lot for your work

**Comment by wigglememore** [comment]
Hi,

Thanks for your hard work on this integration.

I have a Sigenstor 8.0 SP, 8kWh battery, backup gateway and (soon) an AC charger.  I have configured your integration in my Home Assistant instance however everything is showing as unavailable, so I'm not receiving any information.  I have a screenshot from my installer showing the modbus server is active.  I have two devices showing up in my Unifi controller but neither of the IPs seem to work.
![image](https://github.com/user-attachments/assets/8bfbaf91-a90a-4eb0-accf-bc2fcec4cf5f)

Are you able to ping your sigenstor from command prompt?  I'm trying to figure out why homeassistant can't access it.  For one of the IPs I get a request timeout, and with the other I get a destination host unreachable.  I can ping other things on the same VLAN 192.168.50.xx so I don't think(?) it's a network issue, however if others can ping the inverter then it might be...

Many thanks in advance,
Matt

**Reply by micvppl** [reply]
Yup it reply on echo. Maybe try to ping it from your router?
![image](https://github.com/user-attachments/assets/00743e8f-0b3b-46d1-ad75-2a56ba19bd7f)

**Reply by TypQxQ** [reply]
My Sigenergy system (wired) doesn't respond to Ping. You can use the Modbus poll software mentioned above to test the ports.
Please open a new thread for asking for questions.
```
C:\Windows\System32>ping 192.168.1.106

Pinging 192.168.1.106 with 32 bytes of data:
Request timed out.
Request timed out.
Request timed out.
Request timed out.

Ping statistics for 192.168.1.106:
    Packets: Sent = 4, Received = 0, Lost = 4 (100% loss),

C:\Windows\System32>
```
Regards, Andrei

**Comment by Heli-wolf** [comment]
I have been using the Sigenstor EC10 with 16kWh battery for a good week and I installed [https://github.com/TypQxQ/Sigenergy-Home-Assistant-Integration] for the evaluation.
It works very well, but there are so many sensors that are evaluated and made available that I don't have a bit of an overview.
I would like to collect the following data monthly:

- Obtained data from grid
- data delivered to the grid
- generated solar data
- Battery charging
- Battery discharge

Unfortunately I couldn't find these values ​​anywhere, or I overlooked them.
Can someone please help me?

Greetings Wolfgang

**Reply by ZB-42** [reply]
Hi Heli-wolf,

The values you are looking for are all available as daily figures in the Sigen integration. 

For weekly or monthly versions of these values you just need to set up a Utility Meter (found under Helpers) for each of the values you want. Use the existing daily entity as the input and set the period to the appropriate value, in your case monthly. You can then use the newly created utility meter entities to display the data as gauges etc. on your dashboard.  

Further help is available for the use of Utility Meters, just search for it in Home Assistant.

I also have an EC10 with 16kWh batteries and have exactly the same meters set up, both weekly and monthly. They work well.
